Add configurable header extensions, signature name and mod name

jonpas opened the pull request:
**When merged this pull request will:**
- Work on #17 
- Add configurable header extensions
- Add configurable signature name (the middle part between PBO name and ".bisign")
- Add configurable mod name (`@mod` folder name)
- Add configurable key name (`.bikey` and `.biprivatekey` names)
- Add initial replacement patterns for header extensions and signature name properties (will be expanded in the future either with custom handling or [handlebars](https://github.com/sunng87/handlebars-rust) - thus the syntax)
- Change signing to print pretty errors from armake2 instead of panic on failure

Git hash parser can be added in separate PR as utility, otherwise this will get too big. Zip utility will also need it for parsing zip name.

synixebrett commented on docs/json.md:
````diff
@@ -121,3 +121,36 @@ HEMMT will build the specified optionals from the `./optionals` folder.
+```json
+"signame": "my_custom_name"
+```
+
````
It would be nice to put a file tree here showing the final name and place of the file for people that are less familiar with signing.

synixebrett commented on docs/json.md:
````diff
@@ -121,3 +121,36 @@ HEMMT will build the specified optionals from the `./optionals` folder.
+
+```json
+"headerexts": [
+    "author=me"
````
We should probably be doing author ourselves since we already store the author in the project file.

jonpas replied:
No, this is a bad example maybe, but we shouldn't store anything ourselves in there, all it adds is PBO size. We can support author with a replacement pattern though. 

ACE/CBA store git hash here for issue reporting purposes (gets displayed in RPT), that's what I will use it for primarily, but git hash part has yet to be done.

jonpas replied:
Btw, those docs will get updated with possible replacement patterns when I add git hash utility. Example will be better then!

jonpas commented:
Fixed conflicts, also added key name.

Zip name is still missing, but I will add that in the other PR with zip utility, which will actually call it, so we don't have unused functions warnings in the meantime.

I consider this PR done for now.
